StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Elements/PaymentMethodElement/PaymentMethodElement.swift

```diff
@@ -16,15 +16,37 @@
  Other elements can rely on the default implementation provided in this file.
  */
 protocol PaymentMethodElement: Element {
-    /**
-     Modify the params according to your input, or return nil if invalid.
-     - Note: This is called on the Element hierarchy in depth-first search order.
-     */
+    /// Modify the params with default values.
+    ///
+    /// This method is called before `updateParams(params:)` and should be used to populate `params` with any necessary
+    /// default values, these can include values for fields not included in the element hierarchy.
+    func applyDefaults(params: IntentConfirmParams) -> IntentConfirmParams
```

Comment:
This addition ends up having a ton of surface area, so I'd love to learn a little more about it. Right now only cards are passing in a `defaultsApplier` - do we expect different payment methods to want to apply different defaults? Nothing in that closure seems specific to cards, or to the context the element's being created in, so it's not clear to me at a glance the value of this over just initializing the params you pass with the defaults (if the configuration allows for it). Some clarity on that would be super helpful!

Reply:
This will be used by all payment methods, in this PR I'm only adding support for credit cards, but I need a way to inject this into the flow.
The values will depend on the context, they don't in this case because cards are the simplest case, but for example sofort has an special field for country, is not part of the address.
I'm open to alternatives on how to do this, I spend a good chunk of time thinking of a good solution is this is the best I came up with, but happy to discuss more.

Reply:
If there are no concerns or suggestions regarding this I'll resolve the comment